posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -860,16 +895,38 @@ async def run(self, inputs: RunWorkflowInputs) -> Results:
             ),
         )
 
+        # Run the DAG
         run_model_activity_inputs = RunDagActivityInputs(team_id=inputs.team_id, dag=dag)
-        results = await temporalio.workflow.execute_activity(
-            run_dag_activity,
-            run_model_activity_inputs,
-            start_to_close_timeout=dt.timedelta(hours=1),
-            heartbeat_timeout=dt.timedelta(minutes=1),
-            retry_policy=temporalio.common.RetryPolicy(
-                maximum_attempts=1,
-            ),
-        )
+        try:
+            results = await temporalio.workflow.execute_activity(
+                run_dag_activity,
+                run_model_activity_inputs,
+                start_to_close_timeout=dt.timedelta(hours=1),
+                heartbeat_timeout=dt.timedelta(minutes=1),
+                retry_policy=temporalio.common.RetryPolicy(
+                    maximum_attempts=1,
+                ),
+                cancellation_type=temporalio.workflow.ActivityCancellationType.TRY_CANCEL,
+            )
+        except temporalio.exceptions.ActivityError as e:
```

Comment:
Shouldn't we check the exception here? I mean to decide whether we should set the `DataModelingJob` to `Canceled` or `Failed`.

Reply:
Or `Cancelled` with 2 `l`, that confuses me so much.